Docs: Update "Install OpenVINO Runtime on macOS from Archive File" page (#13347)
openvinotoolkit/openvino: docs/install_guides/installing-openvino-from-archive-macos.md

# Install OpenVINO™ Runtime on macOS from an Archive File {#openvino_docs_install_guides_installing_openvino_from_archive_macos}

> With the OpenVINO™ 2022.2 release, you can download and use archive files to install OpenVINO Runtime. The archive files contain pre-built binaries and library files needed for OpenVINO Runtime, as well as sample code for running demos. This page provides instructions showing how to install OpenVINO Runtime using archive files. Check the [Release Notes](https://software.intel.com/en-us/articles/OpenVINO-RelNotes) for more information on updates in the 2022.2 release.

Reviewer: ```suggestion
With the OpenVINO™ 2022.2 release, you can download and use archive files to install OpenVINO Runtime. The archive files contain pre-built binaries and library files needed for OpenVINO Runtime, as well as code samples. This page provides instructions showing how to install OpenVINO Runtime using archive files. Check the [Release Notes](https://software.intel.com/en-us/articles/OpenVINO-RelNotes) for more information on updates in the 2022.2 release.
```

## System Requirements

@sphinxdirective

.. tab:: Operating Systems

>   macOS 10.15

Reviewer 2: let's also add x86, 64 bits

.. tab:: Software Requirements

  * `CMake 3.13 or higher <https://cmake.org/download/>`_ (choose "macOS 10.13 or later"). Add `/Applications/CMake.app/Contents/bin` to path (for default install). 

>   * `Python 3.6 - 3.10 <https://www.python.org/downloads/mac-osx/>`_ (choose 3.6 - 3.10). Install and add to path.

Reviewer 2: we don't support python 3.10 yet

Reviewer 2: I still see that python 3.10 is mentioned

>      * Note that OpenVINO is gradually phasing out the support for Python 3.6. Python 3.7 - 3.10 are recommended.

Reviewer 2: drop 3.10 here. We are going to support it since 22.3

Reviewer: Yeah, this is for master, so it's for 2022.3. Actually this change is done in another PR: https://github.com/openvinotoolkit/openvino/pull/13243

  * Apple Xcode Command Line Tools. In the terminal, run `xcode-select --install` from any directory
  * (Optional) Apple Xcode IDE (not required for OpenVINO™, but useful for development)

.. note::

>    The package may also be installed on other versions of macOS, but only the specific versions listed here are fully validated.

Reviewer 2: can we state that we support all macOS versions starting from 10.15 and explicitly mention x86_64 arch?

>    The package may also be installed on other versions of macOS, but only the specific versions listed here are fully validated.

Reviewer: ```suggestion
   The package can also be installed on macOS x86 versions higher than 10.15, but only the specific versions listed here are fully validated.
```

@endsphinxdirective

## Installing OpenVINO Runtime

### <a name="install-core"></a>Step 1: Install OpenVINO Core Components

1. Open a command prompt terminal window. 
2. Create the `/opt/intel` folder for OpenVINO and switch to it by using the following commands. If the folder already exists, skip the `mkdir` command.
   ```sh
   sudo mkdir /opt/intel
   cd /opt/intel
   ```
   > **NOTE**: The `/opt/intel` path is the recommended folder path for installing OpenVINO. You may use a different path if desired.

> 3. Download the OpenVINO Runtime 2022.2 archive file from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0), extract it, and rename the extracted folder to `openvino_2022.2.0.7713` by using the following commands:

Reviewer: ```suggestion
3. Download the OpenVINO Runtime archive file for macOS from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0), extract it, and rename the extracted folder to `openvino_2022.2.0.7713` by using the following commands:
```

   ```sh
   sudo curl -L https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/m_openvino_toolkit_osx_2022.2.0.7713.af16ea1d79a_x86_64.tgz --output openvino_2022.2.0.7713.tgz
   sudo tar -xf openvino_2022.2.0.7713.tgz

>    sudo mv m_openvino_toolkit_osx_2022.2.0.7713.af16ea1d79a_x86_64 openvino_2022.2.0.7713

Reviewer 2: can we use first two commands w/o sudo and only move from ~/Downloads to /opt with sudo?

   ```

4. For simplicity, it is useful to create a symbolic link as below:
   ```
   sudo ln -s openvino_2022.2.0.7713 openvino_2022
   ```
   > **NOTE**: If you have already installed a previous release of OpenVINO 2022, a symbolic link to the `openvino_2022` folder may already exist. Remove the previous link with `sudo rm openvino_2022`, then re-issue the previous command.

Congratulations, you finished the installation! The `/opt/intel/openvino_2022` folder now contains the core components for OpenVINO. If you used a different path in Step 2, you will find the `openvino_2022` folder there.

> The directory containing the `openvino_2022` folder is also referred as `<INSTALL_DIR>` throughout the OpenVINO documentation.

Reviewer: ```suggestion
The path to the `openvino_2022` directory is also referred as `<INSTALL_DIR>` throughout the OpenVINO documentation.
```

### <a name="set-the-environment-variables"></a>Step 2: Configure the Environment

You must update several environment variables before you can compile and run OpenVINO applications. Open a terminal window and run the `setupvars.sh` script as shown below to temporarily set your environment variables. If your <INSTALL_DIR> is not `/opt/intel/openvino_2022`, use the correct one instead.

```sh
source /opt/intel/openvino_2022/setupvars.sh
```  

If you have more than one OpenVINO™ version on your machine, you can easily switch its version by sourcing the `setupvars.sh` of your choice.

> > **NOTE**: The above command must be re-run every time the terminal is opened. To set up Linux so it automatically runs the command every time a new terminal is opened, open `~/.bashrc` in your favorite editor and add `source /opt/intel/openvino_2022/setupvars.sh` after the last line. Next time when you open a terminal, you will see `[setupvars.sh] OpenVINO™ environment initialized`. Changing `.bashrc` is not recommended when you have multiple OpenVINO™ versions on your machine and want to switch among them.

Reviewer 2: Linux => macOS? Also, macOS has different terminal `zsh` (at least latest versions) and we should use `~/.zshrc` rather than `~/.bashrc`

> > **NOTE**: The above command must be re-run every time the terminal is opened. To set up Linux so it automatically runs the command every time a new terminal is opened, open `~/.bashrc` in your favorite editor and add `source /opt/intel/openvino_2022/setupvars.sh` after the last line. Next time when you open a terminal, you will see `[setupvars.sh] OpenVINO™ environment initialized`. Changing `.bashrc` is not recommended when you have multiple OpenVINO™ versions on your machine and want to switch among them.

Reviewer: ```suggestion
> **NOTE**: The above command must be re-run every time you start a new terminal session. To set up macOS to automatically run the command every time a new terminal is opened, open `~/.zshrc` in your favorite editor and add `source /opt/intel/openvino_2022/setupvars.sh` after the last line. Next time when you open a terminal, you will see `[setupvars.sh] OpenVINO™ environment initialized`. Changing `~/.zshrc` is not recommended when you have multiple OpenVINO versions on your machine and want to switch among them.
```

.. tab:: Software Requirements

  * `CMake 3.13 or higher <https://cmake.org/download/>`_ (choose "macOS 10.13 or later"). Add `/Applications/CMake.app/Contents/bin` to path (for default install). 
  * `Python 3.6 - 3.10 <https://www.python.org/downloads/mac-osx/>`_ (choose 3.6 - 3.10). Install and add to path.

>      * Note that OpenVINO is gradually phasing out the support for Python 3.6. Python 3.7 - 3.10 are recommended.

Reviewer: This is for master, so it's for 2022.3. We can merge this first, and then create a separate PR to remove 3.6 once that's notified by dev team. For the PR to port to 2022.2, I will change it back to 3.9.

  * Apple Xcode Command Line Tools. In the terminal, run `xcode-select --install` from any directory
  * (Optional) Apple Xcode IDE (not required for OpenVINO™, but useful for development)

> 

Reviewer: ```suggestion

.. note::
   The package may also be installed on other versions of macOS, but only the specific versions listed here are fully validated.
```

   ```sh
   sudo mkdir /opt/intel
   cd /opt/intel
   ```
   > **NOTE**: The `/opt/intel` path is the recommended folder path for installing OpenVINO. You may use a different path if desired.

3. Download the OpenVINO Runtime 2022.2 archive file from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0), extract it, and rename the extracted folder to `openvino_2022.2.0.7713` by using the following commands:
   ```sh
   sudo curl -L https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/m_openvino_toolkit_osx_2022.2.0.7713.af16ea1d79a_x86_64.tgz --output openvino_2022.2.0.7713.tgz
   sudo tar -xf openvino_2022.2.0.7713.tgz
   sudo mv m_openvino_toolkit_osx_2022.2.0.7713.af16ea1d79a_x86_64 openvino_2022.2.0.7713
   ```

4. For simplicity, it is useful to create a symbolic link as below:

   ```
   sudo ln -s openvino_2022.2.0.7713 openvino_2022
   ```
   > **NOTE**: If you have already installed a previous release of OpenVINO 2022, a symbolic link to the `openvino_2022` folder may already exist. Remove the previous link with `sudo rm openvino_2022`, then re-issue the previous command.

> Congratulations, you finished installation! The `/opt/intel/openvino_2022` folder now contains the core components for OpenVINO™. When other pages in OpenVINO™ documentation refer to the `<INSTALL_DIR>` directory, this is the folder they're referring to. If you installed OpenVINO™ in a different location, make sure to use that instead.

Reviewer: ```suggestion
Congratulations, you finished the installation! The `/opt/intel/openvino_2022` folder now contains the core components for OpenVINO. If you used a different path in Step 2, you will find the `/openvino_2022` folder there. This path is also mentioned as `<INSTALL_DIR>` throughout the OpenVINO documentation.
```

### <a name="install-core"></a>Step 1: Install OpenVINO Core Components

1. Open a command prompt terminal window. 
2. Create the `/opt/intel` folder for OpenVINO and switch to it by using the following commands. If the folder already exists, skip the `mkdir` command.
   ```sh
   sudo mkdir /opt/intel
   cd /opt/intel
   ```

> 

Reviewer: ```suggestion
```

   > **NOTE**: The `/opt/intel` path is the recommended folder path for installing OpenVINO. You may use a different path if desired.

3. Download the OpenVINO Runtime 2022.2 archive file from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0), extract it, and rename the extracted folder to `openvino_2022.2.0.7713` by using the following commands:
   ```sh
   sudo curl -L https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/m_openvino_toolkit_osx_2022.2.0.7713.af16ea1d79a_x86_64.tgz --output openvino_2022.2.0.7713.tgz
   sudo tar -xf openvino_2022.2.0.7713.tgz
   sudo mv m_openvino_toolkit_osx_2022.2.0.7713.af16ea1d79a_x86_64 openvino_2022.2.0.7713
   ```

> 3. Create a symbolic link to the folder by issuing:

Reviewer: ```suggestion
4. For simplicity, it is useful to create a symbolic link as below:
```

   sudo ln -s openvino_2022.2.0.7713 openvino_2022
   ```
   > **NOTE**: If you have already installed a previous release of OpenVINO 2022, a symbolic link to the `openvino_2022` folder may already exist. Remove the previous link with `sudo rm openvino_2022`, then re-issue the previous command.

> Congratulations, you finished the installation! The `/opt/intel/openvino_2022` folder now contains the core components for OpenVINO. If you used a different path in Step 2, you will find the `/openvino_2022` folder there.

Reviewer: ```suggestion
Congratulations, you finished the installation! The `/opt/intel/openvino_2022` folder now contains the core components for OpenVINO. If you used a different path in Step 2, you will find the `openvino_2022` folder there.
```

> The installation directory is also referred as `<INSTALL_DIR>` throughout the OpenVINO documentation.

Reviewer: ```suggestion
The directory containing the `openvino_2022` folder is also referred as `<INSTALL_DIR>` throughout the OpenVINO documentation.
```

### <a name="install-core"></a>Step 1: Install OpenVINO Core Components

> 1. Open a command prompt terminal window. Create a folder for OpenVINO and move into it by issuing the following commands. If the `/opt/intel` folder already exists, skip the `mkdir` command.

Reviewer: 1. Open a command prompt terminal window. You can use the keyboard shortcut: Ctrl+Alt+T

2. Create the `/opt/intel` folder for OpenVINO and switch to it by using the following commands. If the folder already exists, skip the `mkdir` command.
```

> 1. Open a command prompt terminal window. Create a folder for OpenVINO and move into it by issuing the following commands. If the `/opt/intel` folder already exists, skip the `mkdir` command.

Reviewer: ```suggestion
1. Open a command prompt terminal window. 
2. Create the `/opt/intel` folder for OpenVINO and switch to it by using the following commands. If the folder already exists, skip the `mkdir` command.
```

2. Create the `/opt/intel` folder for OpenVINO and switch to it by using the following commands. If the folder already exists, skip the `mkdir` command.
   ```sh
   sudo mkdir /opt/intel
   cd /opt/intel
   ```

   > **NOTE**: The `/opt/intel` path is the recommended folder path for installing OpenVINO. You may use a different path if desired.

> 2. Download the OpenVINO Runtime 2022.2 archive file from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0), unpack it, and rename the folder to openvino_2022.2.0.7713 by issuing:

Reviewer: ```suggestion
3. Download the OpenVINO Runtime 2022.2 archive file from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0), extract it, and rename the extracted folder to `openvino_2022.2.0.7713` by using the following commands:
```

> 

Reviewer: ```suggestion
```

   sudo tar -xf openvino_2022.2.0.7713.tgz
   sudo mv m_openvino_toolkit_osx_2022.2.0.7713.af16ea1d79a_x86_64 openvino_2022.2.0.7713
   ```

4. For simplicity, it is useful to create a symbolic link as below:

   ```
   sudo ln -s openvino_2022.2.0.7713 openvino_2022
   ```
   > **NOTE**: If you have already installed a previous release of OpenVINO 2022, a symbolic link to the `openvino_2022` folder may already exist. Remove the previous link with `sudo rm openvino_2022`, then re-issue the previous command.

Congratulations, you finished the installation! The `/opt/intel/openvino_2022` folder now contains the core components for OpenVINO. If you used a different path in Step 2, you will find the `/openvino_2022` folder there.

The installation directory is also referred as `<INSTALL_DIR>` throughout the OpenVINO documentation.

### <a name="set-the-environment-variables"></a>Step 2: Configure the Environment

> You must update several environment variables before you can compile and run OpenVINO™ applications. Open a terminal (if it isn't already open) and run the setupvars.sh script as shown below to temporarily set your environment variables. Again, if you installed OpenVINO™ in a folder other than `/opt/intel/openvino_2022`, use that location instead.

Reviewer: ```suggestion
You must update several environment variables before you can compile and run OpenVINO applications. Open a terminal window and run the `setupvars.sh` script as shown below to temporarily set your environment variables. If your <INSTALL_DIR> is not `/opt/intel/openvino_2022`, use the correct one instead.
```

### <a name="install-core"></a>Step 1: Install OpenVINO Core Components

1. Open a command prompt terminal window. 
2. Create the `/opt/intel` folder for OpenVINO and switch to it by using the following commands. If the folder already exists, skip the `mkdir` command.

> 

Reviewer: ```suggestion
```

1. Open a command prompt terminal window. 
2. Create the `/opt/intel` folder for OpenVINO by using the following command. If the folder already exists, skip this command.
   ```sh
   sudo mkdir /opt/intel
   ```
   > **NOTE**: The `/opt/intel` path is the recommended folder path for installing OpenVINO. You may use a different path if desired.

3. Browse to the current user's `Downloads` folder:
   ```sh
    cd <user_home>/Downloads
    ```
    
4. Download the [OpenVINO Runtime archive file for macOS](https://storage.openvinotoolkit.org/repositories/openvino/packages/2022.2/macos/), extract the files, rename the extracted folder and move it to the desired path:
   ```sh
   curl -L https://storage.openvinotoolkit.org/repositories/openvino/packages/2022.2/macos/m_openvino_toolkit_osx_2022.2.0.7713.af16ea1d79a_x86_64.tgz --output openvino_2022.2.0.7713.tgz
   tar -xf openvino_2022.2.0.7713.tgz
   sudo mv m_openvino_toolkit_osx_2022.2.0.7713.af16ea1d79a_x86_64 /opt/intel/openvino_2022.2.0.7713
   ```

5. For simplicity, it is useful to create a symbolic link as below:
   ```
   sudo ln -s openvino_2022.2.0.7713 openvino_2022
   ```

>    > **NOTE**: If you have already installed a previous release of OpenVINO 2022, a symbolic link to the `openvino_2022` folder may already exist. Remove the previous link with `sudo rm openvino_2022`, then re-issue the previous command.

Reviewer: ```suggestion
   > **NOTE**: If you have already installed a previous release of OpenVINO 2022, a symbolic link to the `openvino_2022` folder may already exist. Unlink the previous link with ``sudo unlink openvino_2022``, and then re-run the command above.
```

## Additional Resources

@sphinxdirective

* :ref:`Troubleshooting Guide for OpenVINO Installation & Configuration <troubleshooting guide for install>`

> * Converting models for use with OpenVINO™: :ref:`Model Optimizer Developer Guide <deep learning model optimizer>`

Reviewer: ```suggestion
* Converting models for use with OpenVINO™: :ref:`Model Optimizer User Guide <deep learning model optimizer>`
```
